src/components/inputs/localized-money-input/localized-money-input.spec.js

```diff
@@ -0,0 +1,629 @@
+        onChange={this.handleChange}
+        value={this.state.value}
+        selectedCurrency={this.state.selectedCurrency}
+      />
+    );
+  }
+}
+
+const renderLocalizedMoneyInput = (props, options) =>
+  render(<TestComponent {...props} />, options);
+
+it('should forward data-attributes', () => {
+  const { container } = renderLocalizedMoneyInput({
+    'data-foo': 'bar',
+  });
+  expect(container.querySelector('[data-foo="bar"]')).toBeInTheDocument();
+});
+
+it('should have an HTML name', () => {
+  const { container } = renderLocalizedMoneyInput({
+    name: 'foo',
+    selectedCurrency: 'CAD',
+  });
+  expect(
+    container.querySelector('input[name="foo.CAD.amount"]')
+  ).toBeInTheDocument();
+});
+
+it('should call onBlur when input loses focus', () => {
+  const onBlur = jest.fn();
+  const { container } = renderLocalizedMoneyInput({
+    name: 'foo',
+    selectedCurrency: 'CAD',
+    onBlur,
+  });
+  const input = container.querySelector('input[name="foo.CAD.amount"]');
+  input.focus();
+  expect(input).toHaveFocus();
+  input.blur();
+  expect(input).not.toHaveFocus();
+});
+
+describe('when input is collapsed', () => {
+  it('should not show only the `selectedCurrency`', () => {
+    const { container } = renderLocalizedMoneyInput({
+      name: 'foo',
+      selectedCurrency: 'CAD',
+    });
+    expect(
+      container.querySelector('input[name="foo.CAD.amount"]')
+    ).toBeInTheDocument();
+    expect(
+      container.querySelector('input[name="foo.USD.amount"]')
+    ).not.toBeInTheDocument();
+  });
+  it('should allow changing value of the input', () => {
+    const { container } = renderLocalizedMoneyInput({
+      name: 'foo',
+      selectedCurrency: 'CAD',
+    });
+    const event = { target: { value: '17.34' } };
+    const input = container.querySelector('input[name="foo.CAD.amount"]');
+    fireEvent.focus(input);
+    fireEvent.change(input, event);
+    fireEvent.keyDown(input, { key: 'Enter' });
+    fireEvent.keyUp(input, { key: 'Enter' });
+    expect(input.value).toEqual('17.34');
+  });
+});
+
+describe('when input is expanded', () => {
+  it('should expand and show all currency inputs when `Show all currencies` is clicked', () => {
+    const { container, getByLabelText } = renderLocalizedMoneyInput({
+      name: 'foo',
+      selectedCurrency: 'CAD',
+    });
+    expect(
+      container.querySelector('input[name="foo.USD.amount"]')
+    ).not.toBeInTheDocument();
+    getByLabelText(/show all currencies/i).click();
+    expect(
+      container.querySelector('input[name="foo.USD.amount"]')
+    ).toBeInTheDocument();
+  });
+  it('should allow changing the USD input', () => {
+    const { getByLabelText, container } = renderLocalizedMoneyInput({
+      name: 'foo',
+      selectedCurrency: 'CAD',
+    });
+    getByLabelText(/show all currencies/i).click();
+    const event = { target: { value: '12.98' } };
+    const USDInput = container.querySelector('input[name="foo.USD.amount"]');
```

Comment:
I'd rename this to `usdInput`. It's a regular variable so it should start lowercase. Otherwise the `USDInput.value` access later on looks like we're reading a static property from a class. It's pretty nitpicky, but I wouldn't want this naming pattern to start spreading.